server/api/posts/repository.go

```diff
@@ -29,3 +33,79 @@ func (repository *PostRepository) CreatePost(post model.Post, authorId uint) err
 
 	return repository.database.Exec(query, post.Title, post.Slug, post.Content, post.Status, authorId, createdAt).Error
 }
+
+// FUNCTION TO GET ALL POSTS
+func (reposistory *PostRepository) GetAllPosts() (error, []model.Post) {
+	fmt.Println("STARTING GET/posts request: ")
+	var err error
+	var catID sql.NullInt64
+	var updateTime sql.NullTime
+	//var categoryID sql.NullInt64
+	var posts []model.Post
+	query := "SELECT * FROM posts"
+	results := reposistory.database.Raw(query)
+	if results == nil {
+		err = fmt.Errorf("Fail to fetch posts of user !\nPlease check if posts / user exist in database ! ")
+		return err, nil
+	}
+	rows, err := results.Rows()
+	// Fetch all the posts from DB and assign to the posts slice
+	for rows.Next() {
+		var post model.Post
+		err := rows.Scan(&post.ID, &post.Title, &post.Slug, &post.AuthorID, &catID, &post.Status, &post.CreatedAt, &updateTime, &post.Content)
```

Comment:
**question**: Why is in this database query function so convoluted? Is this required for the `SELECT * FROM posts` query to work?

Reply:
So you want to make it shorter right ? I have just explored on the internet about db.Find(&post) to handle this. If right then I will modify the code.